rosbag2_storage_mcap/src/mcap_storage.cpp

```diff
@@ -252,16 +253,30 @@ MCAPStorage::~MCAPStorage() {
 #ifdef ROSBAG2_STORAGE_MCAP_HAS_STORAGE_OPTIONS
 void MCAPStorage::open(const rosbag2_storage::StorageOptions& storage_options,
                        rosbag2_storage::storage_interfaces::IOFlag io_flag) {
-  open_impl(storage_options.uri, io_flag, storage_options.storage_config_uri);
+  open_impl(storage_options.uri, storage_options.storage_preset_profile, io_flag,
+            storage_options.storage_config_uri);
 }
 #endif
 
 void MCAPStorage::open(const std::string& uri,
                        rosbag2_storage::storage_interfaces::IOFlag io_flag) {
-  open_impl(uri, io_flag, "");
+  open_impl(uri, "", io_flag, "");
 }
 
-void MCAPStorage::open_impl(const std::string& uri,
+static void SetOptionsForPreset(const std::string& preset_profile, McapWriterOptions& options) {
+  if (preset_profile == "fastwrite") {
```

Comment:
I'm also fine with this in theory - I might feel that it was better if this was data-oriented? Maybe a static `unordered_map<std::string, McapWriterOptions>` somewhere that we can just index into? Not a strong requirement though, just a feeling.